### PR TITLE
Use conversion protocol to ensure input types on connection init

### PR DESCRIPTION
**Pull request by witkwski**
Protect borders :shield: earlier then it will be passed to SSL with few extra characters.
Fix double quotes in README.

### CODE REVIEW

**Comment by dankimio** [comment]
@witkwski why do we need to convert certificate data to a string?

**Comment by antnvtkvsk** [comment]
@dankimio for example? `OpenSSL::X509::Certificate#new` expects a string as a certificate and coerces input data with `#to_s` . `OpenSSL::PKey::RSA#new` in that case expects a pair of strings as well (cert and passphrase). So will be better to ensure needed types on connection building instead of catching errors from OpenSSL entrails.

**Comment by witkwski** [comment]
@dankimio for example? OpenSSL::X509::Certificate#new expects a string as a certificate and coerces input data with #to_s . OpenSSL::PKey::RSA#new in that case expects a pair of strings as well (cert and passphrase). So will be better to ensure needed types on connection building instead of catching errors from OpenSSL entrails.

File: lib/houston/connection.rb [hunk 1]
```diff
@@ -26,8 +26,8 @@ def open(uri, certificate, passphrase)
 
     def initialize(uri, certificate, passphrase)
       @uri = URI(uri)
-      @certificate = certificate
-      @passphrase = passphrase
+      @certificate = certificate.to_s
+      @passphrase  = passphrase.to_s
```
Comment (dankimio) [inline lib/houston/connection.rb]: Can you remove extra whitespace here?

Comment (witkwski) [inline lib/houston/connection.rb]: sure :+1:

**Comment by dankimio** [comment]
@witkwski That's a valid point! Thank you for the PR.

**Pull request merged**